src/bvals/boundary_conditions_apk.hpp

```diff
@@ -0,0 +1,90 @@
+//========================================================================================
+// AthenaPK - a performance portable block structured AMR astrophysical MHD code.
+// Copyright (c) 2025, Athena-Parthenon Collaboration. All rights reserved.
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+//! \file boundary_conditions_apk.chpp
+//  \brief AthenaPK specific boundary conditions
+//
+
+#ifndef BVALS_BOUNDARY_CONDITIONS_APK_HPP_
+#define BVALS_BOUNDARY_CONDITIONS_APK_HPP_
+
+#include <memory>
+#include <string>
+#include <vector>
+
+// Parthenon headers
+#include <parthenon/package.hpp>
+
+#include "basic_types.hpp"
+#include "bvals/boundary_conditions_generic.hpp"
+#include "mesh/domain.hpp"
+#include "mesh/mesh.hpp"
+#include "mesh/meshblock.hpp"
+#include "utils/error_checking.hpp"
+
+#include "../main.hpp"
+
+namespace Hydro {
+namespace BoundaryFunction {
+
+using namespace parthenon::package::prelude;
+using parthenon::CoordinateDirection;
+// using parthenon::MeshBlockData;
+// using parthenon::Real;
+using parthenon::BoundaryFunction::BCSide;
+
+template <CoordinateDirection DIR, BCSide SIDE>
+void ReflectBC(std::shared_ptr<MeshBlockData<Real>> &mbd, bool coarse) {
```

Comment:
What does `coarse` mean here?

Reply:
It's called/set from the Parthenon infrastructure for runs with mesh refinement and indicates when the coarse ghost cells of the given block should be filled.